lib/routes/sign.js

```diff
@@ -45,86 +49,104 @@ module.exports = function (log, isA, error, signer, db, domain, metricsContext)
           return reply(error.unverifiedAccount())
         }
 
-        if (publicKey.algorithm === 'RS') {
-          if (!publicKey.n) {
-            return reply(error.missingRequestParameter('n'))
-          }
-          if (!publicKey.e) {
-            return reply(error.missingRequestParameter('e'))
-          }
-        }
-        else { // DS
-          if (!publicKey.y) {
-            return reply(error.missingRequestParameter('y'))
-          }
-          if (!publicKey.p) {
-            return reply(error.missingRequestParameter('p'))
-          }
-          if (!publicKey.q) {
-            return reply(error.missingRequestParameter('q'))
-          }
-          if (!publicKey.g) {
-            return reply(error.missingRequestParameter('g'))
-          }
-        }
-
-        if (!sessionToken.locale) {
-          if (request.app.acceptLanguage) {
-            // Log details to sanity-check locale backfilling.
-            log.info({
-              op: 'signer.updateLocale',
-              locale: request.app.acceptLanguage
-            })
-            db.updateLocale(sessionToken.uid, request.app.acceptLanguage)
-            // meh on the result
-          } else {
-            // We're seeing a surprising number of accounts that don't get
-            // a proper locale.  Log details to help debug this.
-            log.info({
-              op: 'signer.emptyLocale',
-              email: sessionToken.email,
-              locale: request.app.acceptLanguage,
-              agent: request.headers['user-agent']
-            })
-          }
-        }
-        var uid = sessionToken.uid.toString('hex')
-        var deviceId = sessionToken.deviceId ? sessionToken.deviceId.toString('hex') : null
-        var certResult
+        return P.resolve()
+          .then(
+            function () {
+              if (sessionToken.deviceId) {
+                deviceId = sessionToken.deviceId.toString('hex')
+              } else if (! service || service === 'sync') {
+                // Synthesize a device record for Sync sessions that don't already have one
```

Comment:
I think we want to be able to know two things:

* How often do we have to create a placeholder device record, versus devices creating one on their own.  Client-side work would try to drive this number to zero.  I'm about to leave a broader comment on this topic...

* How often do we fail to parse any useful information out of the user-agent header?  As part of that, we should probably log the full user-agent header for debugging purposes.  A `log.info` inside the `truncate()` function might be a good place for this.